sources/api/migration/migrator/src/test.rs

```diff
@@ -0,0 +1,287 @@
+    // are running in their listed order (rather than sorted order as in previous
+    // implementations).
+    manifest.migrations.insert(
+        (Version::new(0, 99, 0), Version::new(0, 99, 1)),
+        vec![FIRST_MIGRATION.into(), SECOND_MIGRATION.into()],
+    );
+    update_metadata::write_file(tuf_indir.join("manifest.json").as_path(), &manifest).unwrap();
+
+    // Create an executable binary that we can use as the 'migration' that migrator will run.
+    // this program will write its name and arguments to a file named results.txt in the
+    // directory that is the parent of `--source-datastore`. results.txt can then be used to see
+    // what migrations ran, and in what order. Note that this program is sensitive to the order
+    // and number of arguments passed. If `--source-datastore` is given at a different position
+    // then the tests will fail and the program will need to be updated.
+    let migration_a = create_test_migration(FIRST_MIGRATION);
+    let migration_b = create_test_migration(SECOND_MIGRATION);
+
+    // Save lz4 compressed copies of this bash script into the tuftool_indir to match the
+    // migration specifications in the manifest.
+    compress(migration_a.as_slice(), &tuf_indir.join(FIRST_MIGRATION));
+    compress(migration_b.as_slice(), &tuf_indir.join(SECOND_MIGRATION));
+
+    // Create and sign the TUF repository.
+    let mut editor = tough::editor::RepositoryEditor::new(root()).unwrap();
+    let long_ago: chrono::DateTime<chrono::Utc> =
+        chrono::DateTime::parse_from_rfc3339("1970-01-01T00:00:00Z")
+            .unwrap()
+            .into();
+    let one = std::num::NonZeroU64::new(1).unwrap();
+    editor
+        .targets_version(one)
+        .targets_expires(long_ago)
+        .snapshot_version(one)
+        .snapshot_expires(long_ago)
+        .timestamp_version(one)
+        .timestamp_expires(long_ago);
+
+    fs::read_dir(tuf_indir)
+        .unwrap()
+        .filter(|dir_entry_result| {
+            if let Ok(dir_entry) = dir_entry_result {
+                return dir_entry.path().is_file();
+            }
+            false
+        })
+        .for_each(|dir_entry_result| {
+            let dir_entry = dir_entry_result.unwrap();
+            editor.add_target(
+                dir_entry.file_name().to_str().unwrap().into(),
+                tough::schema::Target::from_path(dir_entry.path()).unwrap(),
+            );
+        });
+    let signed_repo = editor
+        .sign(&[Box::new(tough::key_source::LocalKeySource { path: pem() })])
+        .unwrap();
+    signed_repo.link_targets(tuf_indir, &targets_path).unwrap();
+    signed_repo.write(&metadata_path).unwrap();
+
+    TestRepo {
+        _tuf_dir: test_repo_dir,
+        metadata_path,
+        targets_path,
+    }
+}
+
+/// Tests the migrator program end-to-end using the `run` function.
+/// The test uses a locally stored tuf repo at `migrator/tests/data/repository`.
```

Comment:
```
/// The test uses a locally stored tuf repo at `migrator/tests/data/repository`.
```

not exactly true anymore since i'm generating the tuf repo